### ISSUE TITLE
"proper" quotient groups

### ISSUE THREAD

**Issue by stevelinton** [gap-system/gap#2378]
There has been recent support discussion about confusion over whether G/G = G/G returns true or false. At present '/' for groups returns a group isomorphic to the quotient and you can recover the homomorphism, but it is not, in fact a group of cosets of the normal subgroup. I'm pretty sure it's been thus ever since GAP 3, but these days we have better technology which we use, for instance, for automorphism groups, so that they are actually groups of automorphisms, even if the real work happens in a nice permutation group behind the scenes.

I am led to wonder if we could do the same for quotient groups -- make them actually groups of cosets of a normal subgroup, but attach enough infrastructure and methods to make them efficient anyway?

**Comment by ThomasBreuer** [comment]
Yes, GAP has always distinguished between cosets and elements of factor groups.
When one deals with a factor group of a given group,
GAP is free to choose a suitable representation of the elements in the factor,
and the natural epimorphism is the connection between the two groups.
I like this concept.
Forcing all factor group elements to carry a coset context around would be an unnecessary burden.
(And this would be increased in common situations where one forms factor groups iteratively.)

I think the case with automorphism groups is different, since one wants the ability to apply elements
of automorphism groups to elements or subgroups of the given group.

It might, however, be worth a try how big the additional overhead is.
Could it be realized as a student's project to introduce an alternative way of realizing factor groups?

**Comment by hulpke** [comment]
For Permutation groups GAP3 had a n implementation of factor groups through cosets. In almost all cases this was unpractically slow. At the transition to GAP4 I thus decides that porting this code was not worth doing, but the natural homomorphism mechanism is faster.

One could add such functionality back as a teaching tool without any harm done, but the shadowing of factor group arithmetic in the whole group will still be a slowdown (and we thus should recommend users to rather just use the homomorphisms.)
If so, would we want to simply multiply representatives or ensure that each coset is always represented by its "chosen" representative?

**Comment by fingolfin** [comment]
While I am sure we could do provide such "coset factor groups", I don't quite understand what advantage this is supposed to bring -- perhaps for teaching purposes, but otherwise, I see only drawbacks: extra overhead (and note that the "nice monomorphism" approach is not perfect (and probably cannot be), and on occasion one encounter's cases where one must work with the "nice" object directly, else the computation gets prohibitively slow); extra complications (each extra layer of artfully arranged abstractions adds to the difficulty to find out what exactly GAP is doing for any given computation); and of course all this for code people use extensively, and where any regression will be felt quite painfully.

In contrast, I can't think of any scenario where I might benefit from this - probably my imagination is too limited, so I'd appreciate examples.

**Comment by fingolfin** [comment]
Of course nowadays there are the hybrid groups by @hulpke which perhaps come a bit closer to this (in some cases). But all in all I don't think we'll act on this, so we may as well close it.
